ban/core/models.py

```diff
@@ -44,11 +44,12 @@ def __str__(self):
 
 
 class Municipality(NamedModel):
+    INSEE_FORMAT = '(2[AB]|\d{2})\d{3}'
     identifiers = ['siren', 'insee']
     resource_fields = ['name', 'alias', 'insee', 'siren', 'postcodes']
     exclude_for_version = ['postcodes']
 
-    insee = db.CharField(length=5, unique=True, format='[\dAB]{2}\d{3}')
+    insee = db.CharField(length=5, unique=True, format=INSEE_FORMAT)
     siren = db.CharField(length=9, format='\d*', unique=True, null=True)
```

Comment:
I see your point.
How it is now:
- if you send a string too short or too long, you have a dedicated message
- if you send a string with the right length, but with wrong format, you'll have a format dedicated message
And we have cases with or without regex, so cases where we'd better keep the length check and dedicated message.
I feel it's good enough, but I'm open to another suggestion :)